### PR TITLE
[AB2D-6145] increase `ab2d-bfd` code coverage

### PR DESCRIPTION
**Pull request by coilysiren**
## 🎫 Ticket

https://jira.cms.gov/browse/AB2D-6145

## 🛠 Changes

- Addresses code smells in tests
- Adds `BFDSearchImplTest` to provide code coverage for `BFDSearchImpl`
- Removes `BFDSearchImpl` from sonarqube excludes

## ℹ️ Context for reviewers

Just this PR won't budge code coverage by much, but it took me forever to write so I want to chunk of my PRs appropriately.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BlueButtonClientR4Test.java [hunk 1]
```diff
@@ -89,15 +89,15 @@ public static void tearDown() {
     }
 
     @Test
-    public void shouldGetEOBFromPatientID() {
```
Comment (coilysiren) [inline ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BlueButtonClientR4Test.java:92]: Its all code smells changes past this point

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    Environment environment = mock(Environment.class);
+    when(environment.getActiveProfiles()).thenReturn(new String[] { activeProfile });
+
+    // Setup classes
+    BfdClientVersions bfdClientVersions = new BfdClientVersions(bfdUrl, httpClient);
+    BFDSearchImpl bfdSearchImpl = new BFDSearchImpl(httpClient, environment, bfdClientVersions);
+
+    // Business logic & assertion(s)
+    IBaseBundle result = bfdSearchImpl.searchEOB(patientId, since, until, pageSize, bulkJobId, version, contractNum);
+    assertNotNull(result);
```
Comment (coilysiren) [inline ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java:82]: I tried to add another assertion here where I assert on the request URL for the HTTP client, but I couldn't get that to work.

This code:

```
    verify(httpClient).execute(new HttpGet("http://localhost:8080/v2/fhir/ExplanationOfBenefit?patient=1&excludeSAMHSA=true&_count=10"));
```

Gave me this error:

```
BFDSearchImplTest > testSearchEOB200() FAILED
    Argument(s) are different! Wanted:
    httpClient.execute(
        GET http://localhost:8080/v2/fhir/ExplanationOfBenefit?patient=1&excludeSAMHSA=true&_count=10 HTTP/1.1
    );
    -> at gov.cms.ab2d.bfd.client.BFDSearchImplTest.testSearchEOB200(BFDSearchImplTest.java:83)
    Actual invocations have different arguments:
    httpClient.execute(
        GET http://localhost:8080/v2/fhir/ExplanationOfBenefit?patient=1&excludeSAMHSA=true&_count=10 HTTP/1.1
    );
    -> at gov.cms.ab2d.bfd.client.BFDSearchImpl.getEOBSFromBFD(BFDSearchImpl.java:90)
        at app//gov.cms.ab2d.bfd.client.BFDSearchImplTest.testSearchEOB200(BFDSearchImplTest.java:83)
```

It looks the same, but says they are different :( probably something to do with comparing two instances of a class with each other.

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  @Test
+  void testSearchEOB200WithFalsyArgs() throws IOException{
```
Comment (coilysiren) [inline ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java:86]: The purpose of this test is just to get code coverage on all of the `if` statements in [`BFDSearchImpl`](https://github.com/CMSgov/AB2D-Libs/blob/main/ab2d-bfd/src/main/java/gov/cms/ab2d/bfd/client/BFDSearchImpl.java)

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    HttpClient httpClient = mock(HttpClient.class);
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class);
+    when(httpClient.execute(any())).thenReturn(response);
+    when(response.getStatusLine()).thenReturn(mock(org.apache.http.StatusLine.class));
+    when(response.getStatusLine().getStatusCode()).thenReturn(HttpStatus.SC_OK);
+    when(response.getEntity()).thenReturn(mock(org.apache.http.HttpEntity.class));
+    String fileContent = Files.readString(Paths.get(filePath));
+    when(response.getEntity().getContent()).thenReturn(new ByteArrayInputStream(fileContent.getBytes()));
+    return httpClient;
+  }
```
Comment (coilysiren) [inline ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java:49]: This took me FOREVER to figure out. I hope its okay 😵

**Review by smirnovaae — APPROVED** [review]

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java (hunk 4 above)
Comment (Sadibhatla) [reply ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java:49]: I am beyond impressed how you found a way to test this , woww!

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java (hunk 2 above)
Comment (Sadibhatla) [reply ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java:82]: its interesting why V1 works but V2 fails. Have this code in place and comment it  with notes that couldnt get it to work for V2 as TODO?

File: build.gradle [hunk 5]
```diff
@@ -167,7 +167,7 @@ subprojects {
 
     sonarqube {
         properties {
-            property 'sonar.coverage.exclusions', "**/BFDSearchImpl.java, **/SQSConfig.java, **/SNSConfig.java"
+            property 'sonar.coverage.exclusions', "**/SQSConfig.java, **/SNSConfig.java"
```
Comment (Sadibhatla) [inline build.gradle:170]: Thanks for taking care of these. Everything looks great to me

File: ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java (hunk 4 above)
Comment (coilysiren) [reply ab2d-bfd/src/test/java/gov/cms/ab2d/bfd/client/BFDSearchImplTest.java:49]: This is what I'm planning on demoing

**Pull request merged**